File: csharp/etl/Etl.cs
public static class Etl
{
    public static Dictionary<string, int> Transform(Dictionary<int, string[]> old)
    {
        Dictionary<string, int> ret = new();
        foreach (KeyValuePair<int, string[]> entry in old)
        {
            foreach (string letter in entry.Value)
            {
                ret[letter.ToLowerInvariant()] = entry.Key;
            }
        }
        return ret;
    }
}

File: php/etl/Etl.php
<?php
// ETL exercise.

declare(strict_types=1);

// Transform the given input array. Swap key = array[value] for lowercase(value) = key
// @param $input: The array to transform.
// @returns: Array of the results of transforming the input.
function transform(array $input): array
{
    $ret = array();
    foreach($input as $key => $values) {
        $points = intval($key);
        foreach($values as $letter) {
            $ret[strtolower($letter)] = $points;
        }
    }
    return $ret;
}
